components/tools/OmeroM/src/createObjectDictionary.m

```diff
@@ -32,10 +32,11 @@
 ip.parse(names, classnames);
 
 % Initialize structure array
-d(1 : nObjects) = struct('name', '', 'class', '', 'Iobject', '', 'delete', '');
+d(1 : nObjects) = struct('name', '', 'class', '', 'Iobject', '', 'delete', '', 'targetObject', '');
 for  i = 1 : nObjects
     d(i).name = names{i};
     d(i).class = ['omero.model.' classnames{i}];
     d(i).Iobject = str2func([d(i).class 'I']);
-    d(i).delete = ['/' classnames{i}];
+    d(i).delete = ['/' classnames{i}];%@deprecated
+    d(i).targetObject = java.util.Hashtable;
```

Comment:
I am not sure this is adding much since the value is independent of the nature of the object. My initial proposal was to store the class name i.e:

```
  d(i).classname = classnames{i}
```

and then use this new field as an argument in replacement of the deprecated `delete`:

```
targetObj.put(objectType.classname, idlist);
```